daffodil-cli/src/main/scala/org/apache/daffodil/Main.scala

```diff
@@ -1355,6 +1355,19 @@ object Main extends Logging {
         log(LogLevel.Error, "%s", e.getMessage())
         1
       }
+      case e: OutOfMemoryError => {
+        System.err.println("""|
+                            |!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+                            |!!             Daffodil ran out of memory!              !!
+                            |!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+                            |
+                            | Try increasing the amount of memory by changing or
+                            | setting the DAFFODIL_JAVA_OPTS environment variable.
```

Comment:
Might as well tell them how to set it i.e., for the OpenJDK JVM the parameter value is -Xmx5G for 5 gigabytes.